### PR TITLE
feat(frontend) : add user and auth store

### PR DESCRIPTION
**Pull request by bdockbockd**

### CODE REVIEW

File: frontend/instagram/src/store/index.ts [hunk 1]
```diff
@@ -1,11 +1,19 @@
 import Vue from "vue";
 import Vuex from "vuex";
+import auth from "@/store/modules/auth";
+// import user from "@/store/modules/user";
+// import VuexPersistence from "vuex-persist";
```
Comment (tirogen) [inline frontend/instagram/src/store/index.ts]: remove comment

File: frontend/instagram/src/store/index.ts [hunk 2]
```diff
@@ -1,11 +1,19 @@
 import Vue from "vue";
 import Vuex from "vuex";
+import auth from "@/store/modules/auth";
+// import user from "@/store/modules/user";
```
Comment (tirogen) [inline frontend/instagram/src/store/index.ts]: remove comment

File: frontend/instagram/src/store/index.ts [hunk 3]
```diff
@@ -1,11 +1,19 @@
...
+// import VuexPersistence from "vuex-persist";
 
 Vue.use(Vuex);
-
-export default new Vuex.Store({
-  state: {},
-  mutations: {},
-  actions: {},
-  modules: {}
+// const vuexLocal = new VuexPersistence<LoginState>({
```
Comment (tirogen) [inline frontend/instagram/src/store/index.ts]: remove comment

File: frontend/instagram/src/store/index.ts [hunk 4]
```diff
@@ -1,11 +1,19 @@
...
+//   storage: window.localStorage,
```
Comment (tirogen) [inline frontend/instagram/src/store/index.ts]: remove comment

File: frontend/instagram/src/store/index.ts [hunk 5]
```diff
@@ -1,11 +1,19 @@
...
+//   modules: ["login"]
```
Comment (tirogen) [inline frontend/instagram/src/store/index.ts]: remove comment

File: frontend/instagram/src/store/index.ts [hunk 6]
```diff
@@ -1,11 +1,19 @@
...
+// });
```
Comment (tirogen) [inline frontend/instagram/src/store/index.ts]: remove comment

File: frontend/instagram/src/store/index.ts [hunk 7]
```diff
@@ -1,11 +1,19 @@
...
+const store = new Vuex.Store({
+  modules: {
+    auth
+  }
+  // plugins: [vuexLocal.plugin]
```
Comment (tirogen) [inline frontend/instagram/src/store/index.ts]: remove comment

File: frontend/instagram/src/store/modules/auth.ts [hunk 8]
```diff
@@ -0,0 +1,141 @@
+// import { cloneDeep } from "lodash";
```
Comment (tirogen) [inline frontend/instagram/src/store/modules/auth.ts]: remove comment

File: frontend/instagram/src/store/modules/auth.ts [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+// import { baseState, baseMutations } from "../state";
```
Comment (tirogen) [inline frontend/instagram/src/store/modules/auth.ts]: remove comment

File: frontend/instagram/src/store/modules/auth.ts [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+  AuthState,
+  LoginCredentials,
+  SignUpCredentials,
+  AuthActions,
+  AuthMutations,
+  AuthGetters
+} from "@/types/auth";
+import { User } from "@/types/user";
+import { StoreOptions } from "vuex";
+import router from "../../router";
```
Comment (tirogen) [inline frontend/instagram/src/store/modules/auth.ts:11]: can we have absolute path?

File: frontend/instagram/src/store/modules/auth.ts [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+        axios.defaults.headers.common[
+          "Authorization"
+        ] = `Bearer ${state.token}`;
+      }
+    }
+  }
+};
+export default store;
+
+// to get state in store
```
Comment (tirogen) [inline frontend/instagram/src/store/modules/auth.ts:135]: remove comment

File: frontend/instagram/src/store/modules/auth.ts [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+// this.$store.state.search.transcribeMessage;
```
Comment (tirogen) [inline frontend/instagram/src/store/modules/auth.ts:136]: remove comment

File: frontend/instagram/src/store/modules/auth.ts [hunk 13]
```diff
@@ -0,0 +1,141 @@
...
+
+// to dispatch
```
Comment (tirogen) [inline frontend/instagram/src/store/modules/auth.ts:138]: remove comment

File: frontend/instagram/src/store/modules/auth.ts [hunk 14]
```diff
@@ -0,0 +1,141 @@
...
+// this.$store.dispatch('search/start')
```
Comment (tirogen) [inline frontend/instagram/src/store/modules/auth.ts:139]: remove comment

File: frontend/instagram/src/store/modules/user.ts [hunk 15]
```diff
@@ -0,0 +1,70 @@
+// import { cloneDeep } from "lodash";
```
Comment (tirogen) [inline frontend/instagram/src/store/modules/user.ts]: remove comment

File: frontend/instagram/src/store/modules/user.ts [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+// import { baseState, baseMutations } from "../state";
```
Comment (tirogen) [inline frontend/instagram/src/store/modules/user.ts]: remove comment

File: frontend/instagram/src/store/modules/user.ts [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+import {
+  UserState,
+  UserGetters,
+  UserMutations,
+  UserActions
+} from "@/types/user";
+import { User } from "@/types/user";
+import { StoreOptions } from "vuex";
+// import router from "../../router";
```
Comment (tirogen) [inline frontend/instagram/src/store/modules/user.ts]: remove comment

File: frontend/instagram/src/store/modules/user.ts [hunk 18]
```diff
@@ -0,0 +1,70 @@
...
+  mutations: {
+    [UserMutations.setUser]: (state: UserState, user: User) => {
+      state.user = user;
+    }
+  },
+  actions: {
+    [UserActions.fetchUser]: async ({ commit, getters }) => {
+      if (getters.getUser) {
+        try {
+          // Patch /user/:id/updateProfile
```
Comment (tirogen) [inline frontend/instagram/src/store/modules/user.ts]: remove comment

File: frontend/instagram/src/store/modules/user.ts [hunk 19]
```diff
@@ -0,0 +1,70 @@
...
+          const response = await axios.get("/user");
+          const responseData: User = response.data;
+          commit(UserMutations.setUser, responseData);
+        } catch {
+          // TODO show error user fetching failed
```
Comment (tirogen) [inline frontend/instagram/src/store/modules/user.ts:36]: remove comment

File: frontend/instagram/src/store/modules/user.ts [hunk 20]
```diff
@@ -0,0 +1,70 @@
...
+        }
+      }
+    },
+    [UserActions.updateProfile]: async (
+      { commit },
+      updatedProfile: Partial<User> | User
+    ) => {
+      try {
+        // Patch /user/:id/updateProfile
```
Comment (tirogen) [inline frontend/instagram/src/store/modules/user.ts:45]: remove comment

File: frontend/instagram/src/store/modules/user.ts [hunk 21]
```diff
@@ -0,0 +1,70 @@
...
+        const response = await axios.patch(
+          `/user/${updatedProfile._id}/updateProfile`
+        );
+        if (response.status === 200) {
+          const responseData: User = response.data;
+          commit(AuthMutations.setUser, responseData);
+        } else {
+          // TODO show error updateProfile failed
```
Comment (tirogen) [inline frontend/instagram/src/store/modules/user.ts:53]: remove comment

File: frontend/instagram/src/store/modules/user.ts [hunk 22]
```diff
@@ -0,0 +1,70 @@
...
+        }
+      } catch {
+        // TODO show error
```
Comment (tirogen) [inline frontend/instagram/src/store/modules/user.ts:56]: remove comment

File: frontend/instagram/src/store/modules/user.ts [hunk 23]
```diff
@@ -0,0 +1,70 @@
...
+      }
+    }
+  }
+};
+export default store;
+
+// to get state in store
```
Comment (tirogen) [inline frontend/instagram/src/store/modules/user.ts:63]: remove comment

File: frontend/instagram/src/store/modules/user.ts [hunk 24]
```diff
@@ -0,0 +1,70 @@
...
+// this.$store.state.search.transcribeMessage;
```
Comment (tirogen) [inline frontend/instagram/src/store/modules/user.ts:64]: remove comment

File: frontend/instagram/src/store/modules/user.ts [hunk 25]
```diff
@@ -0,0 +1,70 @@
...
+
+// to dispatch
```
Comment (tirogen) [inline frontend/instagram/src/store/modules/user.ts:66]: remove comment

File: frontend/instagram/src/store/modules/user.ts [hunk 26]
```diff
@@ -0,0 +1,70 @@
...
+// this.$store.dispatch('search/start')
```
Comment (tirogen) [inline frontend/instagram/src/store/modules/user.ts:67]: remove comment

File: frontend/instagram/src/types/auth.ts [hunk 27]
```diff
@@ -0,0 +1,45 @@
+import { User } from "../types/user";
```
Comment (tirogen) [inline frontend/instagram/src/types/auth.ts:1]: absolute path?

File: frontend/instagram/src/types/auth.ts [hunk 28]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+export interface SignUpCredentials {
+  username: string;
+  password: string;
+  email: string;
+}
+
+export enum AuthGetters {
+  isLogin = "isLogin",
```
Comment (tirogen) [inline frontend/instagram/src/types/auth.ts:23]: ???? why not boolean

File: frontend/instagram/src/types/auth.ts [hunk 29]
```diff
@@ -0,0 +1,45 @@
...
+  isFetchingLogin = "isFetchingLogin",
```
Comment (tirogen) [inline frontend/instagram/src/types/auth.ts:24]: why not boolean?

File: frontend/instagram/src/types/auth.ts (hunk 28 above)
Comment (tirogen) [reply frontend/instagram/src/types/auth.ts:23]: i may be wrong??

File: frontend/instagram/src/types/auth.ts (hunk 29 above)
Comment (tirogen) [reply frontend/instagram/src/types/auth.ts:24]: i may be wrong??

File: frontend/instagram/src/types/user.ts [hunk 30]
```diff
@@ -0,0 +1,40 @@
+export interface User {
+  _id: string;
+  username: string;
+  profilePath: string;
+  bio: string;
+}
+
+// export enum UserGender {
```
Comment (tirogen) [inline frontend/instagram/src/types/user.ts]: remove comment

File: frontend/instagram/src/types/user.ts [hunk 31]
```diff
@@ -0,0 +1,40 @@
...
+//   Male = "male",
```
Comment (tirogen) [inline frontend/instagram/src/types/user.ts]: remove comment

File: frontend/instagram/src/types/user.ts [hunk 32]
```diff
@@ -0,0 +1,40 @@
...
+//   Female = "female"
```
Comment (tirogen) [inline frontend/instagram/src/types/user.ts]: remove comment

File: frontend/instagram/src/types/user.ts [hunk 33]
```diff
@@ -0,0 +1,40 @@
...
+// }
```
Comment (tirogen) [inline frontend/instagram/src/types/user.ts]: remove comment

File: frontend/instagram/src/types/user.ts [hunk 34]
```diff
@@ -0,0 +1,40 @@
...
+export interface UserState {
+  isFetching: boolean;
+  isSuccess: boolean;
+  isError: boolean;
+  user: User | null;
+}
+
+export enum UserGetters {
+  getUser = "getUser"
+  //   getOtherUser = "getOtherUser",
```
Comment (tirogen) [inline frontend/instagram/src/types/user.ts]: remove comment

File: frontend/instagram/src/types/user.ts [hunk 35]
```diff
@@ -0,0 +1,40 @@
...
+  //   getUserById = "getUserById",
```
Comment (tirogen) [inline frontend/instagram/src/types/user.ts]: remove comment

File: frontend/instagram/src/types/user.ts [hunk 36]
```diff
@@ -0,0 +1,40 @@
...
+  //   getFetching = "getFetching"
```
Comment (tirogen) [inline frontend/instagram/src/types/user.ts]: remove comment

**Review by tirogen — CHANGES_REQUESTED** [review]

File: frontend/instagram/src/store/index.ts (hunk 1 above)
Comment (bdockbockd) [reply frontend/instagram/src/store/index.ts]: ok

File: frontend/instagram/src/types/auth.ts (hunk 28 above)
Comment (bdockbockd) [reply frontend/instagram/src/types/auth.ts:23]: It's just the name of  getter parameter. It's actually return `!!state.token` that would check whether token is null or not then return. But this approach need changed in the future. I don't really know that check token would be good for checking login. I thought everytime calling this `isLogin` getter needs to verify at backend again whether this token is valid or not.

Comment (bdockbockd) [reply frontend/instagram/src/types/auth.ts:23]: image

File: frontend/instagram/src/types/auth.ts (hunk 27 above)
Comment (bdockbockd) [reply frontend/instagram/src/types/auth.ts:1]: hmmm? i'm not sure what your point is .

**Pull request merged**